ui/pages/confirmations/hooks/alerts/useTrustSignalAlerts.ts

```diff
@@ -0,0 +1,85 @@
+import { useMemo } from 'react';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { NameType } from '@metamask/name-controller';
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { RowAlertKey } from '../../../../components/app/confirm/info/row/constants';
+import { useConfirmContext } from '../../context/confirm';
+import {
+  useTrustSignal,
+  TrustSignalDisplayState,
+} from '../../../../hooks/useTrustSignals';
+import { SignatureRequestType } from '../../types/confirm';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+export function useTrustSignalAlerts(): Alert[] {
+  const { currentConfirmation } = useConfirmContext();
+  const t = useI18nContext();
+
+  const addressToCheck = useMemo(() => {
+    if (!currentConfirmation) {
+      return null;
+    }
+
+    // For transactions, check the 'to' address
+    if ((currentConfirmation as TransactionMeta)?.txParams?.to) {
+      return (currentConfirmation as TransactionMeta).txParams.to;
+    }
+
+    // For signatures, check the verifying contract if available
+    if ((currentConfirmation as SignatureRequestType)?.msgParams?.data) {
+      try {
+        const data = (currentConfirmation as SignatureRequestType)?.msgParams
+          ?.data;
+        const parsedData = typeof data === 'string' ? JSON.parse(data) : data;
+        if (parsedData?.domain?.verifyingContract) {
+          return parsedData.domain.verifyingContract;
+        }
+      } catch (e) {
+        // Ignore parsing errors
+      }
+    }
+
+    return null;
+  }, [currentConfirmation]);
+
+  const chainId = currentConfirmation?.chainId as string;
+
+  const { state: trustSignalDisplayState } = useTrustSignal(
+    addressToCheck || '',
+    NameType.ETHEREUM_ADDRESS,
+    chainId,
+  );
+
+  return useMemo(() => {
+    if (!addressToCheck || !currentConfirmation) {
```

Comment:
```suggestion
    if (!addressToCheck) {
```

Minor, `addressToCheck` already checks for `currentConfirmation` above